etcd.py

```diff
@@ -267,6 +266,22 @@ def etcd_arguments(self, data_dir, initial_cluster, cluster_state):
             cluster_state
         ]
 
+        # this section handles etcd version specific flags
+        etcdversion = os.environ.get('ETCDVERSION')
+        if etcdversion:
+            etcdversion = list(map(lambda x: int(x), etcdversion.split('.')))
+            # don't try to add additonal flags if we encounter an unexpected version format
+            if len(etcdversion) == 3:
```

Comment:
Not sure if this is worth it TBH. If they decide to do a `3.3.1.2`, this will suddenly break instead of continuing. I'd just check that the length is >= 3, or even drop the check altogether.